src/PVRIptvData.cpp

```diff
@@ -804,39 +804,74 @@ int PVRIptvData::GetFileContents(std::string& url, std::string &strContent)
   return strContent.length();
 }
 
+/* 
+  my_utctime: return non leap seconds since 1970-01-01 00:00 for an UTC timestamp.
+  Other than typical struct tm functions, year is really the year,
+  month starts at 1 (not 0), mday starts at 1 
+
+  Adapted from https://stackoverflow.com/a/31533119
+  Conversion from UTC date to second, signed 64-bit adjustable epoch version.
+  Written by François Grieu, 2015-07-21; public domain. 
+  */
+
+/*
+  Numbering months from March, all months except the one before that origin repeat
+  with a cycle of 5 months totaling 153 days alternating 31 and 30 days, so that,
+  for any month, and without consideration for leap years, the number of days since
+  the previous February can be computed(within a constant) using addition of an appropriate
+  constant, multiplication by 153 and integer division by 5;
+  the correction in days accounting for the rule for leap year on years multiple - of - 100
+  (which by exception to the multiple - of - 4 rules are non - leap except if multiple of 400)
+  can be computed(within a constant) by addition of an appropriate constant, integer division
+  by 100, multiplication by 3, and integer division by 4;
+  we can compute correction for any epoch using the same formula we use
+  in the main computation, and can do this with a macro so that this correction is computed
+  at compilation time.
+  In this adaption, epoch is fixed to 1970
+  */
+
+
+static long long my_utctime(int year, int mon, int mday, int hour, int min, int sec)
```

Comment:
Why not have this as a private static function on the class? Also, the naming does not follow the same format as every other function (i.e. using Pascal case, e.g. MakeTime() or MakeUTCTime()).